### PR TITLE
[UOD-1685] Adding major version to abi versioning

### PR DESCRIPTION
**Pull request by danacos**

### CODE REVIEW

**Review by olivier-norture-ultra — DISMISSED** [review]

File: cmd/dkafka/cdc.go [hunk 1]
```diff
@@ -256,12 +256,13 @@ func executeCdC(cmd *cobra.Command, args []string,
...
-		SchemaVersion:     viper.GetString("cdc-cmd-version"),
-		LocalABIFiles:     localABIFiles,
-		ABICodecGRPCAddr:  viper.GetString("cdc-cmd-abicodec-grpc-addr"),
+		Codec:              viper.GetString("cdc-cmd-codec"),
+		SchemaRegistryURL:  viper.GetString("cdc-cmd-schema-registry-url"),
+		SchemaNamespace:    viper.GetString("cdc-cmd-namespace"),
+		SchemaMajorVersion: viper.GetString("cdc-cmd-major-version"),
+		SchemaVersion:      viper.GetString("cdc-cmd-version"),
+		LocalABIFiles:      localABIFiles,
+		ABICodecGRPCAddr:   viper.GetString("cdc-cmd-abicodec-grpc-addr"),
```
Comment (romain-gilles-ultra) [inline cmd/dkafka/cdc.go:265]: Maybe you should use `GetUint` instead of `GetString` as you are expecting a positive number here ;)

File: app.go [hunk 2]
```diff
@@ -824,10 +829,16 @@ func (msg MessageSchemaGenerator) newNamedSchemaGenOptions(name string, abi *ABI
 	}
 }
 
-func schemaVersion(version string, abiBlockNumber uint32) string {
+func schemaVersion(version string, majorVersion string, abiBlockNumber uint32) string {
 	if version == "" {
-		return fmt.Sprintf("0.%d.0", abiBlockNumber)
+		if majorVersion == "" {
```
Comment (romain-gilles-ultra) [inline app.go:834]: If `majorVersion` becomes a uint as explained below your check becomes `== 0`

**Review by romain-gilles-ultra — CHANGES_REQUESTED** [review]
I think you should use `GetUint` instead of `GetString`

**Review by romain-gilles-ultra — APPROVED** [review]
all good ;)

**Pull request merged**